recipes/nsync/all/conanfile.py

```diff
@@ -0,0 +1,99 @@
+import os
+from conans import ConanFile, CMake, tools
+
+required_conan_version = ">=1.33.0"
+
+
+class NsyncConan(ConanFile):
+    name = "nsync"
+    homepage = "https://github.com/google/nsync"
+    description = "Library that exports various synchronization primitive"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("c", "thread", "multithreading", "google")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False], "shared": [True, False]}
+    default_options = {"fPIC": True, "shared": False}
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+
+    generators = "cmake", "cmake_find_package"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True,
+                  destination=self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["NSYNC_ENABLE_TESTS"] = False
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            patch["base_path"] = self._source_subfolder
```

Comment:
```suggestion
```

Reply:
@prince-chrismc to be on the same page why is this better? To me, we just duplicate the same str constant in both conandata.yml and self._source_subfolder, isn't it better to keep it in one place?

Reply:
It's less code in the recipe? But this is mostly because it's our conventional and keeps everything aligned

Reply:
Ok, but to me +1 line of code in receipt is much better then implicit dependency between conandata.yml:/patches/base_path and self._source_subfolder